ui-materialui: Add support for `<Edit emptyWhileLoading>` (#10230)
marmelab/react-admin: docs/Edit.md

);
```

> ## `emptyWhileLoading`

Reviewer: don't forget to add it to the list of props earlier in the doc

> By default, `<Edit>` renders its child component even before the `dataProvider.getOne()` call returns. And default layout components (`<Datagrid>` and `<SimpleList>`) return null when the data is loading. If you use a custom layout component instead, you'll have to handle the case where the `data` is not yet defined.

Reviewer: Why mention `Datagrid` and `SimpleList` here? We're not in a list page.

> But if you use a custom child component that expects the record context to be defined, your component will throw an error. For instance, the following will fail on load with a "ReferenceError: data is not defined" error:

Reviewer: The first sentence says more or less the same as the previous one... not clear. The second mentions an error about undefined data but there is no data in the snippet.

);
```

## `emptyWhileLoading`

By default, `<Edit>` doesn't render its child component even before the `dataProvider.getOne()` call returns, but it returns its [`aside`](#aside), [`actions`](#actions) and [`title`](#title) components.

> If you use one of those components customized with a record, your component will throw an error. For instance, the following will fail on load with a "ReferenceError: record is not defined" error:

Reviewer 2: ```suggestion
If a component you use in those props relies on the record, it will throw an error. For instance, the following will fail on load with a "ReferenceError: record is not defined" error:
```

```jsx
import { Edit, useEditContext } from 'react-admin';
import { Typography } from '@mui/icons-material/Star';

const AsideComponent = () => {
    const { record } = useEditContext();
    return (
        <Typography>
            <i>{record.title}</i>, by {record.author} ({record.year})
        </Typography>
    );
}

const BookEdit = () => (
    <Edit aside={AsideComponent}>
        {/* ... */}
    </Edit>
);
```

You can handle this case by getting the `isPending` variable from the [`useEditContext`](./useEditContext.md) hook:

```jsx
const AsideComponent = () => {
    const { record, isPending } = useEditContext();
    if (isPending) return null;
    return (
        <Typography>
            <i>{record.title}</i>, by {record.author} ({record.year})
        </Typography>
    );
}
```

> The `<Edit emptyWhileLoading>` prop provides a convenient shortcut for that use case. When enabled, `<Edit>` won't render its child until `data` is defined.

Author: ```suggestion
The `<Edit emptyWhileLoading>` prop provides a convenient shortcut for that use case. When enabled, `<Edit>` won't render its `aside`, `actions` and `title` components until `record` is defined.
```
